tmail-backend/jmap/extensions/src/main/java/com/linagora/tmail/james/jmap/StandaloneEventAttendanceRepository.java

```diff
@@ -0,0 +1,124 @@
+package com.linagora.tmail.james.jmap;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Predicate;
+
+import jakarta.inject.Inject;
+import jakarta.mail.Flags;
+
+import org.apache.james.core.Username;
+import org.apache.james.mailbox.MailboxSession;
+import org.apache.james.mailbox.MessageIdManager;
+import org.apache.james.mailbox.MessageManager;
+import org.apache.james.mailbox.SessionProvider;
+import org.apache.james.mailbox.exception.MailboxException;
+import org.apache.james.mailbox.model.FetchGroup;
+import org.apache.james.mailbox.model.MailboxId;
+import org.apache.james.mailbox.model.MessageId;
+import org.apache.james.mailbox.model.MessageResult;
+import org.reactivestreams.Publisher;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import reactor.core.publisher.Mono;
+
+public class StandaloneEventAttendanceRepository implements EventAttendanceRepository {
+    private static final Logger LOGGER = LoggerFactory.getLogger(StandaloneEventAttendanceRepository.class);
+    private static final List<String> EVENT_ATTENDANCE_FLAGS = List.of("$accepted", "$rejected", "$tentativelyaccepted");
+
+    private final MessageIdManager messageIdManager;
+    private final SessionProvider sessionProvider;
+
+    @Inject
+    public StandaloneEventAttendanceRepository(MessageIdManager messageIdManager, SessionProvider sessionProvider) {
+        this.messageIdManager = messageIdManager;
+        this.sessionProvider = sessionProvider;
+    }
+
+    @Override
+    public Publisher<AttendanceStatus> getAttendanceStatus(Username username, MessageId messageId) {
+        MailboxSession systemMailboxSession = sessionProvider.createSystemSession(username);
+
+        return getFlags(messageId, systemMailboxSession)
+            .map(this::getAttendanceStatusFromFlags);
+    }
+
+    @Override
+    public Publisher<Void> setAttendanceStatus(Username username, MessageId messageId,
+                                               AttendanceStatus attendanceStatus) {
+        MailboxSession systemMailboxSession = sessionProvider.createSystemSession(username);
+
+        return getFlags(messageId, systemMailboxSession)
```

Comment:
We are getting flags more than twice before REPLACing them.

Two things about it:
 - it is innefficient
 - And get-then-set do not commute so it will be not concurrent friendly

I prefer us to add the one flag that interest us and remove the two others that do not interest us.